client/clientfactory.go

```diff
@@ -156,8 +156,10 @@ func (cf *rpcClientFactory) NewHistoryClientWithTimeout(timeout time.Duration) (
 		return cf.newHistoryThriftClient(clientKey)
 	}
 
+	defaultMessageSize := cf.rpcFactory.GetMaxMessageSize()
 	client := history.NewClient(
 		cf.numberOfHistoryShards,
+		cf.dynConfig.GetIntProperty(dynamicconfig.GRPCMaxSizeInByte, defaultMessageSize),
```

Comment:
nit: Does it mean the dynamic value we set should always be less than the `MaxMessageSize` from rpc factory? If so, shall we do a check to ensure that?